parser.py

```diff
@@ -106,9 +106,15 @@ def parse(filename):
                 elif tag == 'HUSB':
                     current['husband'] = namedict[args]
                     spousedict[namedict[args]] = args
+                    current['husbandID'] = args
```

Comment:
There is no reason to save the id AND the name, we should be using one to lookup the other. For instance, what if at the end of parse we also want to display the husband's birthdate. We definitely wouldn't want to store that, we would get into the position where we are repeating the same data in every possible other unrelated structure.